lib/go-atscfg/atsdotrules.go

```diff
@@ -37,6 +51,12 @@ type ATSDotRulesOpts struct {
 	HdrComment string
 }
 
+// MakeATSDotRules constructs a '50-ats.rules' file for the given server with
+// the given parameters and header comment content.
+//
+// WARNING: this will panic if parameters with the Names 'Drive_Prefix',
```

Comment:
This isn't true. It generates fine if any or all params are missing.

```
func TestMakeATSDotRulesNoFiles(t *testing.T) {
	server := makeGenericServer()
	serverProfile := "myProfile"
	server.Profile = &serverProfile

	hdr := "myHeaderComment"

	serverParams := []tc.Parameter{}

	cfg, err := MakeATSDotRules(server, serverParams, &ATSDotRulesOpts{HdrComment: hdr})
	if err != nil {
		t.Fatal(err)
	}
	txt := cfg.Text

	testComment(t, txt, hdr)

	if count := strings.Count(txt, "\n"); count != 1 { // one line for each drive letter, plus 1 comment
		t.Errorf("expected one line for a comment and no others (with no parameters), actual: '%v' count %v", txt, count)
	}
}
```

```
$ go test -failfast -v -run TestMakeATSDotRulesNoFiles
=== RUN   TestMakeATSDotRulesNoFiles
--- PASS: TestMakeATSDotRulesNoFiles (0.00s)
PASS
ok      github.com/apache/trafficcontrol/lib/go-atscfg  0.487s
```

Reply:
Actually that isn't entirely true. There's a known bug, if you have a Prefix but no Letters it generates `KERNEL=="ra", OWNER="ats"`, but that's a data bug anyway, so it hasn't been a huge priority to fix.

But it never panics, AFAIK. If you can demonstrate a panic, definitely file a bug.

Reply:
My bad, I thought `(map[string]interface{}{})["anything"]` would panic like `([]interface{}{})[100]` does.

It's fixed, but I've probably made similar mistakes elsewhere, given my lack of familiarity with the package. I would greatly appreciate if you'd look over the `lib/go-atscfg` changes yourself, since you're the best candidate for knowing whether descriptions are accurate.